File: NovaCore-1_17-Plus-Commons/src/net/novauniverse/novacore1_17plus/shared/BaseVersionIndependentUtilImplementation1_17Plus.java
package net.novauniverse.novacore1_17plus.shared;

import org.bukkit.entity.Player;

import net.novauniverse.spigot.version.shared.v1_16plus.BaseVersionIndependentUtilImplementation1_16Plus;
import net.novauniverse.spigot.version.shared.v1_16plus.DyeColorToMaterialMapper;
import net.zeeraa.novacore.spigot.abstraction.VersionIndependantLoader;

public abstract class BaseVersionIndependentUtilImplementation1_17Plus extends BaseVersionIndependentUtilImplementation1_16Plus {
	public BaseVersionIndependentUtilImplementation1_17Plus(VersionIndependantLoader loader, DyeColorToMaterialMapper colorToMaterialMapper) {
		super(loader, colorToMaterialMapper);
	}

	@Override
	public int getPlayerPing(Player player) {
		return player.getPing();
	}

	@Override
	public void sendTabList(Player player, String header, String footer) {
		player.setPlayerListHeaderFooter(header, footer);
	}
}

File: NovaCore-1_17-Plus-Commons/src/main/kotlin/net/novauniverse/novacore1_17plus/shared/BaseVersionIndependentUtilImplementation1_17Plus.kt
package net.novauniverse.novacore1_17plus.shared

import net.novauniverse.spigot.version.shared.v1_16plus.BaseVersionIndependentUtilImplementation1_16Plus
import net.novauniverse.spigot.version.shared.v1_16plus.DyeColorToMaterialMapper
import net.zeeraa.novacore.spigot.abstraction.VersionIndependentLoader
import org.bukkit.entity.Player

abstract class BaseVersionIndependentUtilImplementation1_17Plus(loader: VersionIndependentLoader, colorToMaterialMapper: DyeColorToMaterialMapper) : BaseVersionIndependentUtilImplementation1_16Plus(loader, colorToMaterialMapper) {
	override fun getPlayerPing(player: Player): Int {
		return player.ping
	}

	override fun sendTabList(player: Player, header: String, footer: String) {
		player.setPlayerListHeaderFooter(header, footer)
	}
}